src/components/results/sensitivityAnalysis/sensitivity-analysis-result.js

```diff
@@ -0,0 +1,148 @@
+
+    if (nOrNkIndex === 1) {
+        returnedTable.push(
+            makeColumn(false, 'ContingencyId', 'contingencyId', intl)
+        );
+    }
+
+    const suffix1 = 'In' + ['kW', 'kA', 'kV'][functionTypeIndex];
+    const suffix = suffix1 + (nOrNkIndex !== 1 ? '' : 'BeforeContingency');
+
+    returnedTable.push(
+        makeColumn(true, 'ValRef' + suffix, 'functionReference', intl)
+    );
+    returnedTable.push(makeColumn(true, 'Delta' + suffix, 'value', intl));
+
+    if (nOrNkIndex === 1) {
+        returnedTable.push(
+            makeColumn(true, 'ValRef' + suffix1, 'functionReferenceAfter', intl)
+        );
+        returnedTable.push(
+            makeColumn(true, 'Delta' + suffix1, 'valueAfter', intl)
+        );
+    }
+
+    return returnedTable;
+}
+
+function makeRows(resultRecord) {
+    // Replace NaN values by empty string
+    return resultRecord.map((obj) => {
+        return Object.entries(obj).reduce((acc, [key, value]) => {
+            acc[key] = value === 'NaN' ? '' : value;
+            return acc;
+        }, {});
+    });
+}
+
+const SensitivityAnalysisResult = ({
+    result,
+    nOrNkIndex,
+    sensiToIndex,
+    onSortChanged,
+}) => {
+    const gridRef = useRef(null);
+    const intl = useIntl();
+    const theme = useTheme();
+
+    const columns = useMemo(
+        () => buildTableColumns(nOrNkIndex, sensiToIndex, intl),
+        [nOrNkIndex, sensiToIndex, intl]
+    );
+
+    const rows = useMemo(() => makeRows(result), [result]);
+
+    const defaultColDef = useMemo(
+        () => ({
+            suppressMovable: true,
+            sortable: true,
+            unSortIcon: true,
+        }),
+        []
+    );
+
+    const getRowStyle = useCallback(
+        (params) => {
+            if (params?.data?.elementId) {
+                return {
+                    backgroundColor: theme.selectedRow.background,
+                };
+            }
+        },
+        [theme.selectedRow.background]
+    );
+
+    const onGridReady = useCallback((params) => {
+        if (params.api) {
+            params.api.sizeColumnsToFit();
+        }
+    }, []);
+
+    useEffect(() => {
+        if (gridRef.current) {
+            gridRef.current.api?.refreshCells();
+        }
+    }, [result]);
+
+    return (
+        <div style={{ flexGrow: 1 }}>
+            {!result?.length ? (
+                <FormattedMessage id="NoResults" />
+            ) : (
+                <CustomAGGrid
+                    ref={gridRef}
+                    rowData={rows}
+                    columnDefs={columns}
+                    getRowStyle={getRowStyle}
+                    defaultColDef={defaultColDef}
+                    onGridReady={onGridReady}
+                    onSortChanged={onSortChanged}
+                />
+            )}
+        </div>
+    );
+};
+
+SensitivityAnalysisResult.defaultProps = {
+    result: null,
+    nOrNkIndex: 0,
+    sensiToIndex: 0,
+};
```

Comment:
I don't think we should define default value for required props

Reply:
yes, I removed the required prop. thank you